src/Atmos/Model/bc_energy.jl

```diff
@@ -203,3 +199,95 @@ function atmos_energy_normal_boundary_flux_second_order!(
     # NOTE: difference from design docs since normal points outwards
     fluxᵀn.energy.ρe -= C_h * ρ_avg * normu_int⁻_tan * (MSE - MSE_int)
 end
+
+"""
+    NishizawaEnergyFlux(fn) :: EnergyBC
+Calculate the net inward energy flux across the boundary following Nishizawa and Kitamura (2018).
+Return the flux (in W m^-2).
+"""
+struct NishizawaEnergyFlux{FNTM} <: EnergyBC
+    fn_T_and_q_tot::FNTM
+end
+function atmos_energy_boundary_state!(
+    nf,
+    bc_energy::NishizawaEnergyFlux,
+    atmos,
+    args...,
+) end
+function atmos_energy_normal_boundary_flux_second_order!(
+    nf,
+    bc_energy::NishizawaEnergyFlux,
+    atmos,
+    fluxᵀn,
+    n⁻,
+    state⁻,
+    diffusive⁻,
+    hyperdiffusive⁻,
+    aux⁻,
+    state⁺,
+    diffusive⁺,
+    hyperdiffusive⁺,
+    aux⁺,
+    t,
+    state_int⁻,
+    diffusive_int⁻,
+    aux_int⁻,
+)
+    FT = eltype(state⁻)
+    # Interior point
+    u_int⁻ = state_int⁻.ρu / state_int⁻.ρ
+    u_int⁻_tan = projection_tangential(atmos, aux_int⁻, u_int⁻)
+    normu_int⁻_tan = norm(u_int⁻_tan)
+    q_tot_int =
+        atmos.moisture isa DryModel ? FT(0) :
+        state_int⁻.moisture.ρq_tot / state_int⁻.ρ
+    x_in = MArray{Tuple{3}, FT}(FT[
+        normu_int⁻_tan,
+        aux_int⁻.moisture.θ_v,
+        q_tot_int,
+    ])
+
+    # Boundary point
+    u⁻ = state⁻.ρu / state⁻.ρ
+    u⁻_tan = projection_tangential(atmos, aux⁻, u⁻)
+    normu⁻_tan = norm(u⁻_tan)
+    T, q_tot = bc_energy.fn_T_and_q_tot(atmos, state⁻, aux⁻, t)
+    x_s = MArray{Tuple{3}, FT}(FT[normu⁻_tan, T, q_tot])
+
+    ## Initial guesses for MO parameters, these should be a function of state.
+    LMO_init = FT(1.0)
+    u_star_init = FT(0.1)
+    th_star_init = T
+    qt_star_init = FT(1e-5)
+    MO_param_guess = MArray{Tuple{4}, FT}(FT[
+        LMO_init,
+        u_star_init,
+        th_star_init,
+        qt_star_init,
+    ])
+
+    # Roughness and interior lengths
+    z_0 = FT(0.1)
```

Comment:
Should z_0 be an input of the function?

Reply:
This is just a simple test to showcase that surface fluxes work, these things will be added later on.

Reply:
I ended up adding it :)